contrib/mesos/pkg/queue/delay.go

```diff
@@ -138,6 +139,29 @@ func (q *DelayQueue) Pop() interface{} {
 	}, nil)
 }
 
+func finishWaiting(cond *sync.Cond, waitFinished <-chan struct{}) {
```

Comment:
agree. I'd really like to have something cleaner, but this should stop the
bleeding.

On Fri, Apr 29, 2016 at 8:46 AM, Dr. Stefan Schimanski <
notifications@github.com> wrote:

> In contrib/mesos/pkg/queue/delay.go
> <https://github.com/kubernetes/kubernetes/pull/24916#discussion_r61570968>
> :
>
> > @@ -138,6 +139,29 @@ func (q *DelayQueue) Pop() interface{} {
> >  	}, nil)
> >  }
> >
> > +func finishWaiting(cond *sync.Cond, waitFinished <-chan struct{}) {
>
> this is pretty ugly. But I don't know a better solution right now.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/24916/files/a74573277f81565b3aeff8c5d4bfe2a3110c1ccd#r61570968>
>